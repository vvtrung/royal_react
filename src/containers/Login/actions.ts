import { createActions } from 'redux-actions';

import { ActionTypes } from './constants';

export const { login } = createActions({
  [ActionTypes.LOGIN]: payload => payload,
});
