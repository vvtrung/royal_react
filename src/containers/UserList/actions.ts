import { createActions } from 'redux-actions';

import { User } from 'types';

import { ActionTypes } from './constants';

export const { getUsers, saveUsers } = createActions({
  [ActionTypes.GET_USERS]: payload => payload,
  [ActionTypes.SAVE_USERS]: (users: User) => ({ users }),
});
