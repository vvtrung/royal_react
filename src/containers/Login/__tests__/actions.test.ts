import { login } from '../actions';

describe('Login Actions', () => {
  it('should create login action', () => {
    expect(
      login({ username: 'admin', password: '123123qwe' }),
    ).toMatchSnapshot();
  });
});
