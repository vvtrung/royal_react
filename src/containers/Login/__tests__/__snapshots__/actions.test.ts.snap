// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Actions should create login action 1`] = `
Object {
  "payload": Object {
    "password": "123123qwe",
    "username": "admin",
  },
  "type": "LOGIN",
}
`;
